sql/core/src/main/scala/org/apache/spark/sql/RelationalGroupedDataset.scala

```diff
@@ -708,7 +708,7 @@ private[sql] object RelationalGroupedDataset {
     case expr: NamedExpression => expr
     case a: AggregateExpression if a.aggregateFunction.isInstanceOf[TypedAggregateExpression] =>
       UnresolvedAlias(a, Some(Column.generateAlias))
-    case u: UnresolvedFunction => UnresolvedAlias(expr, None)
+    case e if !e.resolved => UnresolvedAlias(expr, None)
```

Comment:
why is this change?

Reply:
I changed the `AggregateExpression` path to `case a: AggregateExpression => UnresolvedAlias(a, Some(Column.generateAlias))` but still kept the `case _ if !expr.resolved => UnresolvedAlias(expr, None)` because in the `plus_one(sum_udf(col("v1")))` case an unresolved `PythonUDF` is passed here.